### PR TITLE
Add more logs and attempt to fix CI Loop instability: Build and Pack NuGet randomly fails

### PR DESCRIPTION
**Pull request by licanhua**
This PR is attempted to but will not completely fix bug #3444.
From what I see, problem is timeout in 'Run Desktop Integration Tests', and here are some possible reasons:
1. app crashes because of assert command
2. app crashes in RunTest
3. test case is waiting for metro bundle's response
4. when one of the tests fail, the test doesn't continue and I don't know which test is still running.

I'm trying to:
1. add blame flag for vstest, and expect it provides information like: `The active Test Run was aborted because the host process exited unexpectedly while executing following test(s):
RNTesterIntegrationTests::IntegrationTestHarness`
2. RunConfiguration.TestSessionTimeout=120000 to avoid wait for ever
3. try...catch...exception and convert the exception to test failure with the current_exception_diagnostic_information message.
4. `List Desktop Integration Tests` to list all tests when vstest failed
5. Increase the wait time to 2 minutes to exclude the problem of readiness of metro bundle
6. re-run the test if it failed.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3492)

### CODE REVIEW

File: vnext/IntegrationTests/TestRunner.cpp [hunk 1]
```diff
@@ -13,6 +13,8 @@
 #include <objbase.h>
 #include <locale>
 
+#include <boost/exception/all.hpp>
```
Comment (JunielKatarn) [inline vnext/IntegrationTests/TestRunner.cpp]: Out of curiosity, what does this fix / enhance?

**Comment by JunielKatarn** [comment]
> This PR is attempted to but will not completely fix bug #3444.
> From what I see, problem is timeout in 'Run Desktop Integration Tests', and here are some possible reasons:
> 
> 1. app crashes because of assert command
> 2. app crashes in RunTest

All the instances I have observed are test cancellations due to timeout.

> 3. test case is waiting for metro bundle's response

The bundler and WebSocket servers do very light-weight tasks and transmit relatively small content.
The only scenario I think this could happen is if the bundler itself crashed and was no longer available.

> 4. when one of the tests fail, the test doesn't continue and I don't know which test is still running.

Not sure about this. If a test fails, the next one gets executed, in principle. The root issue is most likely a hang (not a explicit failure, but an infinite wait).

**Comment by licanhua** [comment]
> All the instances I have observed are test cancellations due to timeout.
In my local machine, if the app crashed, the vstest.console is always spinning.

File: vnext/IntegrationTests/TestRunner.cpp (hunk 1 above)
Comment (licanhua) [reply vnext/IntegrationTests/TestRunner.cpp]: I didn't see any real exception in testing yet, but I hope to provide more logs when exception happens.
At the beginning, I was trying the catch the assert which results in a crash.
Finally I found assert crash aborted the process and I can't catch it.
The blame would helps with the assert crash.

File: vnext/IntegrationTests/TestRunner.cpp [hunk 2]
```diff
@@ -52,76 +54,81 @@ TestResult TestRunner::RunTest(
...
-    instance->AttachMeasuredRootView(move(appName));
-
-    // Failure on instance creation.
-    if (TestStatus::Failed == result.Status)
-      return result;
-
-    AwaitEvent(functionCalled, result);
-
-    instance->DetachRootView();
   }
```
Comment (asklar) [inline vnext/IntegrationTests/TestRunner.cpp]: we wouldn't get a stack trace here though? can we capture the stack trace from the exception?

Comment (licanhua) [reply vnext/IntegrationTests/TestRunner.cpp]: Stack trace is not what I target yet.
exception is just suspicious which may contribute the timeout. Here catching all exception is just help to narrow down the problem.
If I see this error, I will provide more logs like catch(std::exception) and log the stack.

---
In reply to: [337866067](https://github.com/microsoft/react-native-windows/pull/3492#discussion_r337866067) [](ancestors = 337866067)

**Pull request merged**